scripting/conn-id-ctx.rst

```diff
@@ -0,0 +1,103 @@
+
+.. _script-conn-id-ctx:
+
+==============================
+Use of :zeek:see:`conn_id_ctx`
+==============================
+
+.. versionadded:: 8.0
+
+.. note::
+
+   We’re still iterating on patterns for working with the new pluggable
+   connection keys and :zeek:see:`conn_id_ctx` instances.
+   If you have feedback or run into limitations for your use-cases, please reach out!
+
+In certain deployments, Zeek will receive network traffic from different
+network segments that may have overlapping IP ranges.
+For example, host 10.0.0.37 in segment A and host 10.0.0.37 in segment B
+share the same IP address, but represent different systems.
+The terminology used within Zeek is that IP addresses or connections
+are observed in different "contexts".
+
+
+Commonly, ethernet layers may contain information that makes it possible to discriminate
+between different network segments. Concretely, VLAN tagging is one such approach.
+Segments can also be distinguished by Virtual Network Identifiers (VNIs)
+in case of UDP-based tunnels like VXLAN or Geneve.
+
+
+Since Zeek 8.0, this information can be extracted by
+:ref:`plugin-provided connection key implementations <connkey-plugin>`
+and included into Zeek's core connection tracking. Further, plugins
+:zeek:keyword:`redefine <redef>` :zeek:see:`conn_id_ctx` with additional
+fields to expose the extracted information used for connection tracking to
+the Zeek scripting layer.
```

Comment:
```suggestion
Since version 8.0, Zeek can extract these contexts through
:ref:`plugin-provided connection key implementations <connkey-plugin>`
and include them into its core connection tracking. Such plugins then also
:zeek:keyword:`redefine <redef>` :zeek:see:`conn_id_ctx` with additional
fields to expose the context values to the Zeek scripting layer.
```